### PR TITLE
feat(compaction): notify meta when hash mapping changes

### PR DESCRIPTION
**Pull request by soundOfDestiny**
## What's changed and what's your intention?

notify meta when hash mapping changes

### CODE REVIEW

File: src/meta/src/stream/scheduler.rs [hunk 1]
```diff
@@ -156,6 +156,8 @@ where
             return Err(InternalError("fragment has no actor".to_string()).into());
         }
 
+        self.hash_mapping_manager.set_need_consolidation(true);
```
Comment (hzxa21) [inline src/meta/src/stream/scheduler.rs:159]: Currently we don't have a way to distinguish between fragment creation v.s. fragment reconfiguration in scheduler. Once  we have one, I assume set_need_consolidation should only happen on fragment reconfiguration, right?

Comment (soundOfDestiny) [reply src/meta/src/stream/scheduler.rs:159]: > Currently we don't have a way to distinguish between fragment creation v.s. fragment reconfiguration in scheduler. Once we have one, I assume set_need_consolidation should only happen on fragment reconfiguration, right?

agree that set_need_consolidation should only happen on fragment reconfiguration

**Review by hzxa21 — APPROVED** [review]

**Review by xx01cyx — COMMENTED** [review]
1. Why use "consolidation" instead of "compaction"? 🤔
2. I think some comments are necessary, but it seems that the PR has been merged 🤣 Might add some comments about the functions and the newly-added field in `HashMappingManager` later.

**Comment by soundOfDestiny** [comment]
> > * Why use "consolidation" instead of "compaction"? 🤔
> > * I think some comments are necessary, but it seems that the PR has been merged 🤣 Might add some comments about the functions and the newly-added field in `HashMappingManager` later.
> 
> * to make it different from typical compaction
> * will add in the foreseeable future

no need now :-)

**Pull request merged**